### PR TITLE
docs: react docs arrange

### PR DESCRIPTION
**Pull request by MadCcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

1. 重排研发文档
2. 更新定制主题文档

image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |       -    |
| 🇨🇳 Chinese |      -     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 1dd109c</samp>

This pull request improves the documentation and the code style of the antd React library. It adds new features and fixes to the sandpack and useMenu components, and updates the styling and customization options for the sandpack code block. It also reorganizes the documentation files into different groups, adds the toc and tag props to the frontmatter, and adds a new server-side rendering guide. It removes the unused motion demo files and adjusts the import order of some modules.

### <samp>🤖 Generated by Copilot at 1dd109c</samp>

*  Add `group` and `tag` props to the frontmatter of some markdown files to move them to different groups and show version tags in the menu ([link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-1224d5ca5233fdeba67ba2e2dc6fa3e615905253f35dc132c4b4b86f51f9eab4R6), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-5a985421a0686b8aaf62a2c17c5f66df4bff969ab5bb43b6ab73df640f48874cL2-R6), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-8e29d6fdf5d52f6033a313bef228b6678b1459496af652cf24c69c1b6472a825L2-R4), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-30aba60ae4a347b7452e2a4a66635f4be7e498b360267b963f87c35772c92aaeL2-R5), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-23ec5ad4a63587155920a6d487b591db3a1710ce906a0783eee8039d88e4200dL2-R4), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L2-R5), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-be1faf3d418140212756aac28b89919752b536ca91a6125cb0ae3ddd761efd72L2-R5), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-5cf8ac316a6ed4e65b3b1b24392776c72e475be3e1f54eb062faf7d71d387b47L2-R5), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-d075c199614f1086c766a83821a217cc494c698e0ac6b7c8cbfe8b53602bedd4R1-R333), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-e506cd45013ef022ee1e0e5d855ee37b6d91d4019e2f32a73c758df5441b108eL2-R4), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-2f16c7c378b53633917dedacb793879596836c46978bdaf2c152bc76b97dd627L2-R4))
*  Modify the `useMenu` function in `.dumi/hooks/useMenu.tsx` to insert and render the changelog and React doc items with the version tags and fix the layout issue of the tags inside the links ([link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-97b7cacb6d4bd1445cc4d73ae072d3aac76d65d4c10eb29328c6cc8aef1d6649L4-R4), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-97b7cacb6d4bd1445cc4d73ae072d3aac76d65d4c10eb29328c6cc8aef1d6649L18), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-97b7cacb6d4bd1445cc4d73ae072d3aac76d65d4c10eb29328c6cc8aef1d6649L35-R34), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-97b7cacb6d4bd1445cc4d73ae072d3aac76d65d4c10eb29328c6cc8aef1d6649L43-R58), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-97b7cacb6d4bd1445cc4d73ae072d3aac76d65d4c10eb29328c6cc8aef1d6649L106-R121), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-97b7cacb6d4bd1445cc4d73ae072d3aac76d65d4c10eb29328c6cc8aef1d6649L112-R127), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-97b7cacb6d4bd1445cc4d73ae072d3aac76d65d4c10eb29328c6cc8aef1d6649L134-R151))
*  Support different variants of the sandpack code block, such as 'sandpackdark', by checking the `lang` prop of the `SourceCode` node in `.dumi/rehypeAntd.ts` and adding the `dark` prop to the sandpack component in `.dumi/theme/builtins/Sandpack/index.tsx`. Also, include and import the `index.css` file for the sandpack example ([link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-f0d95b6f380e7a072830e7026f73450a2056d161a2b265783fa556338e677525L71-R77), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-eb1983e6c2de0da4944ef9eb6428a1850e347a11b36c1b8f18dfacada229c76cL26-R26), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-eb1983e6c2de0da4944ef9eb6428a1850e347a11b36c1b8f18dfacada229c76cR34), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-eb1983e6c2de0da4944ef9eb6428a1850e347a11b36c1b8f18dfacada229c76cR40-R48), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-eb1983e6c2de0da4944ef9eb6428a1850e347a11b36c1b8f18dfacada229c76cL65-R75), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-eb1983e6c2de0da4944ef9eb6428a1850e347a11b36c1b8f18dfacada229c76cR94-R102))
*  Allow the user to disable the table of contents by setting the `toc` prop to false in the frontmatter of the markdown file and adding a condition to render the affix and anchor components in `.dumi/theme/slots/Content/index.tsx` ([link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-1533ad458dddb5cacbf24062cc5d262ca684a57e2754bf4777f24f37411d5f82L208-R235))
*  Remove the motion demo code block from `docs/react/customize-theme.en-US.md` as it is not related to the theme customization and change the import order of some modules to follow the alphabetical order ([link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-e12721e5a68f1eaac8d53987399a4d6eb6c7e4e6bb59fed4ffb73c7967079090L24-R25), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-e12721e5a68f1eaac8d53987399a4d6eb6c7e4e6bb59fed4ffb73c7967079090L51-R52), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-e12721e5a68f1eaac8d53987399a4d6eb6c7e4e6bb59fed4ffb73c7967079090L72-R73), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-e12721e5a68f1eaac8d53987399a4d6eb6c7e4e6bb59fed4ffb73c7967079090L103-L105), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-e12721e5a68f1eaac8d53987399a4d6eb6c7e4e6bb59fed4ffb73c7967079090L117-R115), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-e12721e5a68f1eaac8d53987399a4d6eb6c7e4e6bb59fed4ffb73c7967079090L149-R147), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-e12721e5a68f1eaac8d53987399a4d6eb6c7e4e6bb59fed4ffb73c7967079090L178-R177), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-e12721e5a68f1eaac8d53987399a4d6eb6c7e4e6bb59fed4ffb73c7967079090L355-R353), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-e12721e5a68f1eaac8d53987399a4d6eb6c7e4e6bb59fed4ffb73c7967079090L368-R368), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-e12721e5a68f1eaac8d53987399a4d6eb6c7e4e6bb59fed4ffb73c7967079090L514-R515))
*  Delete the `docs/react/demo/motion.tsx` and `docs/react/demo/motion.md` files as they are no longer used ([link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-1ea2f2efdfbfbfa9a0b1caeb9d984cec3c2e22b703530c1d7695513459f117f9), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-5e268624dd3f40258240caaa845d3647355dc3e7cdec4ac148f71417c2f452f7))
*  Add a new section about the shadow DOM scenario to `docs/react/compatible-style.zh-CN.md` to explain how to use the `StyleProvider` component with the `container` prop ([link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-5a985421a0686b8aaf62a2c17c5f66df4bff969ab5bb43b6ab73df640f48874cR127-R146))
*  Change the import order of some modules to follow the alphabetical order and simplify the arrow function return statement in `docs/react/faq.zh-CN.md` ([link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-d24a54fb235784b2741dc29b3c94dbb9fc2396e2837d68a5ae3f36dd5705d524L134-R136), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-d24a54fb235784b2741dc29b3c94dbb9fc2396e2837d68a5ae3f36dd5705d524L299-R309))
*  Change the import order of some modules to follow the alphabetical order in `docs/react/use-with-create-react-app.zh-CN.md` ([link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-2f16c7c378b53633917dedacb793879596836c46978bdaf2c152bc76b97dd627L54-R57), [link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-2f16c7c378b53633917dedacb793879596836c46978bdaf2c152bc76b97dd627L73-R76))
*  Add the whole content of `docs/react/server-side-rendering.zh-CN.md` to explain how to use antd with server-side rendering and different style extraction methods ([link](https://github.com/ant-design/ant-design/pull/44029/files?diff=unified&w=0#diff-d075c199614f1086c766a83821a217cc494c698e0ac6b7c8cbfe8b53602bedd4R1-R333))

### CODE REVIEW

File: CHANGELOG.en-US.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
 ---
 order: 6
-title: Change Log
+title: Changelog
 toc: false
 timeline: true
+tag: vVERSION
```
Comment (afc163) [inline CHANGELOG.en-US.md:6]: vVERSION？

**Comment by li-jia-nan** [comment]
大佬用 CP 切换 algorithm 的功能还需要强调一下，之前在掘金看到有人吐槽这个在文档里找不到，不知道在什么犄角旮旯才看到的……感觉需要放在显眼的地方

**Comment by MadCcc** [comment]
> 大佬用 CP 切换 algorithm 的功能还需要强调一下，之前在掘金看到有人吐槽这个在文档里找不到，不知道在什么犄角旮旯才看到的……感觉需要放在显眼的地方

https://github.com/ant-design/ant-design/pull/44029/files#diff-7238b66024192aa79bff8e41259edd3a2c30155c7422b58c53e6fc63d679a4afR59

File: CHANGELOG.en-US.md (hunk 1 above)
Comment (MadCcc) [reply CHANGELOG.en-US.md:6]: VERSION 是动态值。试了很多模版语法，都会被 dumi 截胡，应该是 yml 转换时就被替换了

**Review by yoyo837 — APPROVED** [review]

**Pull request merged**